hack/verify-go-modules.sh

```diff
@@ -26,7 +26,22 @@ set -o pipefail
 
 REPO_ROOT=$(cd "$(dirname "${BASH_SOURCE[0]}")/.." && pwd)
 
-mapfile -t DIRS < <(find "${REPO_ROOT}" -name go.mod -print0 | xargs -0 dirname)
+# This is the list of directories that host the individual kcp go modules.
+#
+#   KEEP IT UP TO DATE and SORTED!
```

Comment:
You might consider using `go list` from the root to a) get a full list of modules and b) build the DAG for dependencies, so that keeping the list in order and updated is not a human problem.

Reply:
Yeah I was also thinking about that too, but then we have only a couple of these nested go modules in the repo so I wasn't sure if that warrants all the code complexity that would bring. If I have time I could push this in a separate commit to see the diff and decide?

Reply:
@stevekuznetsov I've pushed https://github.com/kcp-dev/kcp/pull/3373/commits/4a0a660c0523e2b2e1b6690eecc538c3e07dde1c that generates the list and sorts it. Does that look more or less reasonable?